RoT/Protocol/RFCs/0000-Capabilities2.md

```diff
@@ -0,0 +1,140 @@
+    the bus). They are interpreted as in Device Capabilities, too.
+  - Bits [5:4] must be zero.
+- "RSA Support", `0x01`. Six bit record. Optional; if not present, implies that
+  RSA encryption is not supported.
+  - Bits [2:0] are the supported key strengths. The bits are interpreted as in
+    the Device Capabilities message.
+  - Bits [5:3] must be zero.
+- "ECDSA Support", `0x02`. Six bit record. Optional; if not present, implies that
+  ECDSA encryption is not supported. (NOTE: Cerberus currently does not specify
+  any curves, so we don't do so here, either.)
+  - Bits [2:0] are the supported key strengths. The bits are interpreted as in
+    the Device Capabilities message.
+  - Bits [5:3] must be zero.
+- "AES Support", `0x03`. Six bit record. Optional; if not present, implies that
+  AES encryption (and, thus, confidentiality) is unsupported.
+  - Bits [2:0] are the supported key strengths, and are as in the Device
+    Capabilities message.
+  - Bits [5:3] must be zero.
+
+The payload sizes and message timeouts are not included, since those are
+intended to be relegated to the transport (per RFC-TODO). The "PFM support",
+"Policy support", and "Firmware Protection" bits are currently not given
+specified meaning, they are not included in the proposal.
+
+Finally, we propose deprecating the Capabilities message. If a device receives a
+Device Capabilities message, it MAY interpret it as a Negotiation message with
+the relevant fields and react accordingly, but may reject it if the equivalent
+Negotiation message would be unsuitable. It may also reject all Device
+Capabilities messages altogether. Clients do not need to send a Device
+Capabilities message at all, unless they wish to be compatible with existing
+devices that require it.
+
+# Specification Changelist
+
+This proposal adds an entire new section to the list of challenge messages,
+adding Negotiation under the current Device Capabilities message. It also adds
+it to the list of commands at the top of the section.
+
+Additionally, the Device Capabilities message is marked as deprecated, with the
+suggested remediation strategy described above.
+
+# Alternatives Considered
+
+We could add another capabilities message every time we need to negotiate a new
+capability. For example, if we were to add a new certificate format, we would
+add a new message for negotiating that, or add a bit to the GET_CERTIFICATES
+message. However, this would significantly complicate the handshake, and require
+burning command message bytes far more often than is strictly necessary.
+
+If we adopt transport agonosticism, several fields in the current Device
+Capabilities message become meaningless, and would need to be zeroed out but
+still sent over the wire.
+
+# Future Work
+
+The encoding described a above leaves us with significant leeway to add even
+more capabilities contents. The flags byte, for example, can be used to
```

Comment:
Which is the flags byte?  Does this refer to 'reserved'?

Reply:
Yup. Name changed while I was editing, oops.